lib/media/streaming_engine.js

```diff
@@ -2159,8 +2191,44 @@ shaka.media.StreamingEngine = class {
                   reference, box));
     }
 
-    if (hasEmsg || shouldParsePrftBox) {
-      parser.parse(segment);
+    if (shouldFixTimestampOffset) {
+      parser
+          .box('moof', shaka.util.Mp4Parser.children)
+          .box('traf', shaka.util.Mp4Parser.children)
+          .fullBox('tfdt', async (box) => {
+            goog.asserts.assert(
+                box.version != null,
+                'TFDT is a full box and should have a valid version.');
+
+            const parsedTFDT = shaka.util.Mp4BoxParsers.parseTFDTInaccurate(
+                box.reader, box.version);
+
+            const baseMediaDecodeTime = parsedTFDT.baseMediaDecodeTime;
+
+            // In case the time is 0, it is not updated
+            if (!baseMediaDecodeTime) {
+              return;
+            }
+            goog.asserts.assert(typeof(timescale) == 'number',
+                'Should be an number!');
+
+            const scaledMediaDecodeTime = -baseMediaDecodeTime / timescale;
+
+            const comparison1 = Number(mediaState.lastTimestampOffset) || 0;
+
+            if (comparison1 < scaledMediaDecodeTime) {
+              const lastAppendWindowStart = 0 +
```

Comment:
Are you just doing this "0 +" stuff to fit the variable on the next line? Or is it to trick the compiler or something?

Reply:
Yes, but i can use an assert. I'll change it